### ISSUE TITLE
typeorm v0.2.9 MySQL - Ignoring invalid configuration option passed to Connection

### ISSUE THREAD

**Issue by rmuchall** [typeorm/typeorm#3127]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.9` (or put your version here)

When using typeorm v0.2.9 on MySQL the following error message occurs in the console when establishing a connection:
`
Ignoring invalid configuration option passed to Connection: acquireTimeout. This is currently a warning, but in future versions of MySQL2, an error will be thrown if you pass an invalid configuration options to a Connection
`

**Comment by rmuchall** [comment]
Reverting to typeorm v0.2.8 "fixes" the issue.

**Comment by bmccann36** [comment]
I am using v0.2.8 and just started experiencing this issue

**Comment by bmccann36** [comment]
I was inadvertently using 2.9
I got confused because I was actually getting a different error that had to do with ssl config and I was conflating the two but they are caused by different things.
downgrading the version fixes the error mentioned in this thread
thanks

**Comment by zaksesti-ns8** [comment]
I think there is a related issue maybe with options in general.  I also experienced the above.   
In addition I'm not able to connect with `mysql@2.6.4` and `typeorm@0.2.9`.   
`typeorm@0.2.8` is OK.   
` Access denied for user ''@'172.17.0.1' (using password: NO)`  

Options:
```
  const options: ConnectionOptions = {
    type: 'mysql',
    url: 'mysql://localhost:3306/',
    username: 'root',
    database: 'protect',
    entities: [Post],
    migrations: [`${__dirname}/migrations/**/*.{t,j}s`],
    migrationsRun: true,
    timezone: 'Z',
    logging: true,
    dropSchema: true,
  };
```

**Comment by josecolella** [comment]
I'm also experiencing this issue after upgrade to latest typeorm

**Comment by NoelBaron** [comment]
I received this error as well using 0.2.9. Although it doesn't prohibit a connection today, it looks like the MySQL team will reject connections soon.

`Ignoring invalid configuration option passed to Connection: acquireTimeout. This is currently a warning, but in future versions of MySQL2, an error will be thrown if you pass an invalid configuration options to a Connection`

**Comment by pleerock** [comment]
Everybody reported this issue - can you please provide more information about what you are doing? Maybe your connection options? Do you use `url` option in connection option?

**Comment by NoelBaron** [comment]
We're using `mysql2`. 
```
createConnection({
  type: "mysql",
  host: process.env.CONNECTION_HOST,
  port: Number(process.env.CONNECTION_PORT),
  username: process.env.CONNECTION_USERNAME,
  password: process.env.CONNECTION_PASSWORD,
  database: process.env.CONNECTION_SOURCE,
  entities: ["dist/models/*.js"],
  synchronize: false,
  logging: false
})
```

**Comment by rmuchall** [comment]
I pull connection options from environment variables:

```
TYPEORM_CONNECTION  = mysql
TYPEORM_HOST        = localhost
TYPEORM_PORT        = 3306
TYPEORM_USERNAME    = user
TYPEORM_PASSWORD    = password
TYPEORM_DATABASE    = mydb
TYPEORM_SYNCHRONIZE = true
TYPEORM_LOGGING     = false
TYPEORM_DROP_SCHEMA = false

const connectionOptions: ConnectionOptions = await getConnectionOptions();
```

**Comment by pleerock** [comment]
can somebody provide a small repo with reproduction of this problem?

**Comment by varsis** [comment]
This issue happens because the key is included in the config if undefined keys are omitted it would fix the issue. 

https://github.com/typeorm/typeorm/blob/master/src/driver/mysql/MysqlDriver.ts#L779-L799

**Comment by varsis** [comment]
The corresponding code in mysql2 
https://github.com/sidorares/node-mysql2/blob/master/lib/connection_config.js#L64-L74

**Comment by josecolella** [comment]
@varsis What's the workaround?

**Comment by MetaiR** [comment]
any idea?

**Comment by varsis** [comment]
When I have a chance I will do a pr it’s an easy fix.

**Comment by ryanvade** [comment]
I'm seeing this in `0.2.12` with `mysql2 1.6.4`.  
```typescript
export default async function init() {
  return await createConnection({
    type: "mysql",
    host: process.env.MYSQL_HOST,
    port: Number(process.env.MYSQL_PORT),
    username: process.env.MYSQL_USERNAME,
    password: process.env.MYSQL_PASSWORD,
    database: process.env.MYSQL_DATABASE,
    entities: [/**/]
  });
}
```

**Comment by kiesel** [comment]
I believe this warning was introduced in node-mysql2 in this commit: https://github.com/sidorares/node-mysql2/commit/023342f78024171b69bc489216d6a3fa5d4dcb88#diff-a8492c5e53a8e7843fe4a95fd1d04213 (which is included since v1.6.2 of mysql2).

The `acquireTimeout` setting itself was never supported or understood by `mysql2` - but it is a valid option in `mysql`. Probably the code should be changed so, so the setting will only be considered as an option when it actually carries a value (so, when using mysql2 as the backend, you could omit the option from the configuration and avoid that particular warning).

**Comment by kiesel** [comment]
This was fixed by #3753 - please close the bugreport:)
